model/task_lifecycle.go

```diff
@@ -753,22 +756,32 @@ type RestartTaskResults struct {
 // or due to the system, respectively
 func RestartFailedTasks(opts RestartTaskOptions) (RestartTaskResults, error) {
 	results := RestartTaskResults{}
-	if opts.OnlyRed && opts.OnlyPurple {
-		opts.OnlyRed = false
-		opts.OnlyPurple = false
+	if !opts.IncludeTestFailed && !opts.IncludeSysFailed && !opts.IncludeSetupFailed {
+		opts.IncludeTestFailed = true
+		opts.IncludeSysFailed = true
+		opts.IncludeSetupFailed = true
 	}
-	tasksToRestart, err := task.Find(task.ByTimeStartedAndFailed(opts.StartTime, opts.EndTime))
+	failedTasks, err := task.Find(task.ByTimeStartedAndFailed(opts.StartTime, opts.EndTime))
```

Comment:
Can we just leverage the database for this operation, and build a $in query based on the options?

Reply:
yep, fixed